internal/pkg/override/cdk.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package override
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/spf13/afero"
+)
+
+// Executable is the interface to wrap os/exec calls.
+type Executable interface {
+	LookPath(file string) (string, error)
+	Command(name string, args ...string) *exec.Cmd // TODO(efe): Command should set cmd.Dir = rootAbsPath.
+}
+
+// CDK is an Overrider that can transform a CloudFormation template with the Cloud Development Kit.
+type CDK struct {
+	rootAbsPath string // Absolute path to the overrides/ directory.
+
+	out  io.Writer  // Writer for any os/exec calls.
+	fs   afero.Fs   // OS file system.
+	exec Executable // For testing os/exec calls.
+}
+
+// NewCDK instantiates a new CDK Overrider with the file system pointing ot the overrides/ dir.
+func NewCDK(root string, stdout io.Writer, fs afero.Fs, exec Executable) *CDK {
+	return &CDK{
+		rootAbsPath: root,
+		out:         stdout,
+		fs:          fs,
+		exec:        exec,
+	}
+}
+
+// Install runs "npm install" insides the overrides/ directory to pull the latest dependencies.
+func (cdk *CDK) Install() error {
+	if _, err := cdk.exec.LookPath("npm"); err != nil {
+		return &errNPMUnavailable{parent: err}
+	}
+
+	cmd := cdk.exec.Command("npm", "install")
+	cmd.Stdout = cdk.out
+
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf(`run %q: %w`, cmd.String(), err)
+	}
+	return nil
+}
+
+// Override writes the template body under dir/.build/in.yml and runs "cdk synth"
+// to generate and return a transformed template.
+// Override should be invoked after Install.
+func (cdk *CDK) Override(body []byte) ([]byte, error) {
+	buildPath := filepath.Join(cdk.rootAbsPath, ".build")
+	if err := cdk.fs.MkdirAll(buildPath, fs.ModeDir); err != nil {
+		return nil, fmt.Errorf("create %s directory to store the CloudFormation template body: %w", buildPath, err)
+	}
+	inputPath := filepath.Join(buildPath, "in.yml")
+	if err := afero.WriteFile(cdk.fs, inputPath, body, 0644); err != nil {
+		return nil, fmt.Errorf("write CloudFormation template body content at %s: %w", inputPath, err)
+	}
+
+	// We assume that a node_modules/ dir is present with the CDK downloaded after running "npm install".
+	// This way clients don't need to install the CDK toolkit separately.
+	cmd := cdk.exec.Command(filepath.Join("node_modules", "aws-cdk", "bin", "cdk"), "synth", "--no-version-reporting")
```

Comment:
override.ts file creation is not a part of this PR right? CDK synth will be applied on the override.ts file which will be created using cloudformation_include module right?

Reply:
Right, I'm planning to send a separate PR for each of the following sections:
1.  Add to `templates/` the scaffold code for the CDK: `override.ts` and the other files.
2. Add a `ScaffoldWithCDK` func in the `override` pkg
3. Update `svc_package.go` and `svc_deploy.go` to use `override`
4. Create `svc_override.go`, `job_override.go` and so on to use `override.ScaffoldWithCDK`.